py3status/modules/usbguard.py

```diff
@@ -0,0 +1,350 @@
+        )
+        for placeholder in available_placeholders:
+            if placeholder in placeholders:
+                self.placeholders[placeholder] = None
+        self.placeholders['usbguard_id'] = None
+
+        self.killed = threading.Event()
+        UsbguardListener(self).start()
+
+    def _set_policy(self, action, index):
+        if action and index and index != 'sep':
+    
+            targets = {'allow': 0, 'block': 1, 'reject': 2}
+
+            self.py3.log('# data')
+            self.py3.log(self.data)
+            self.py3.log('# index')
+            self.py3.log(index)
+            self.py3.log('# action')
+            self.py3.log(action)
+            self.py3.log('# usbguard data')
+            self.py3.log(self.usbguard_data)
+            self.py3.log('# all devices')
+            self.py3.log(self.all_devices)
+            
+            # notifications
+            if self.format_notification:
+                format_notification = self.data[index]
+                format_notification['action'] = action
+                format_notification = self.py3.safe_format(
+                    self.format_notification, format_notification
+                )
+                notification = self.py3.get_composite_string(format_notification)
+                self.py3.notify_user(
+                    notification,
+                    title='USBGuard',
+                    icon='/usr/share/icons/hicolor/scalable/apps/usbguard-icon.svg',
+                )
+
+            # apply policy
+            if 'usbguard_id' in self.data and index != 'sep':
+                self.proxy.applyDevicePolicy(
+                    self.data['usbguard_id'], targets[action], self.is_permanent
+                )
+
+    def _manipulate_all_devices(self, data):
+        format_all_devices = []
+        format_device_separator = self.py3.safe_format(self.format_device_separator)
+        self.py3.composite_update(format_device_separator, {'index': 'sep'})
+
+        for device in data:
+            for x in self.thresholds:
+                if x in data[device]:
+                    self.py3.threshold_get_color(data[device][x], x)
+
+            device_formatted = self.py3.safe_format(
+                self.format_all_devices, data[device]
+            )
+
+            # add usbguard_id
+            self.py3.composite_update(device_formatted, {'usbguard_id': data[device]['usbguard_id']})
+
+            format_all_devices.append(device_formatted)
+
+        for index, device in sorted(enumerate(format_all_devices), reverse=False):
+            format_all_devices[index] = device
+
+        format_all_devices = self.py3.composite_join(
+            format_device_separator, format_all_devices
+        )
+
+        return format_all_devices
+
+    def usbguard(self):
+        if self.error:
+            self.py3.error(str(self.error), self.py3.CACHE_FOREVER)
+
+        self.all_devices = self._get_all_devices()
```

Comment:
I assume you want `or` at end with something else to simplify some things below.